### DISCUSSION TITLE
Export to Markdown inserts extra line breaks

### DISCUSSION THREAD

**Discussion by leehach** [freeplane/freeplane#612] [Help] [answered]
Before I report this as a bug, I want to understand the expected behavior.

I use Markdown in notes. I've been doing this for over 10 years, and wrote a very simple exporter to generate Markdown files. It appears that sometime in the last year Freeplane added this as a core feature. (There is currently no documentation on exporting at <https://docs.freeplane.org/#/user-documentation/working-w-mindmap-files?id=export>, so not sure when this happened or how it supposed to work, but there's discussion of it in issue #133.)

After upgrading to Freeplane 1.10.3, when I use Markdown export, every line is being treated as a paragraph, so it appears in the text document with extra line breaks.

There are several things about this I don't have enough information about:

1. What's the expected behavior? Are extra line breaks supposed to be there?
2. I don't know how to tell whether the note is using plain text or rich text, and whether that makes a difference in the export. There appears to be a way to set the ***node core*** to plain text, but the ***note*** appears to always have an HTML rich text version "under the hood".
3. I'm also not sure if the exporter is using the new Freeplane functionality or my old converter.

# Examples

Note:

```
1. Item
5. Item
6. Item
```

Old export:

```
1. Item
2. Item
3. Item
```

New export:

```
1. Item

2. Item

3. Item
```

Note:

```
This is a paragraph.

This is another paragraph.

This is a third paragraph.
```

Old export:

```
This is a paragraph.

This is another paragraph.

This is a third paragraph.
```

New export:

```
This is a paragraph.



This is another paragraph.



This is a third paragraph.
```

This is particularly bad with code listings.

Note

~~~
Here is some sample code:

```
x = 1
print(x)
```
~~~

Old export:

~~~
Here is some sample code:

```
x = 1
print(x)
```
~~~

New export:

~~~
Here is some sample code:



```

x = 1

print(x)

```

~~~

**Comment by macmarrum** [comment, marked as answer]
I believe you refer to `File->Export map…` and the converter [mm2markdown.xsl](https://github.com/freeplane/freeplane/blob/1.10.x/freeplane/src/external/resources/xslt/mm2markdown.xsl)

I have little experience with using the functionality, but as far as I understand, `mm2markdown.xsl` works on raw xml, so it matters how the information is represented there.

What I can see is that the output in an exported .md file depends on **note's** _Content Type_. When it's set to **Markdown**, there are no additional lines, as compared to _Content Type_ **Standard**.

BTW, Markdown can be set as follows:
- `View->Controls->Tool panel->Format->Content types->For note` = `Markdown`

Looking at the raw xml in an .mm file, here's what I can see in case of **Markdown**

```xml
<node TEXT="Markdown" ID="ID_1780456534"><richcontent TYPE="NOTE" CONTENT-TYPE="plain/markdown">
    <text>1. Item
5. Item
6. Item</text>
</richcontent>
</node>
```

and here's a snippet representing **Standard**

```xml
<node TEXT="Standard" ID="ID_282441170"><richcontent TYPE="NOTE" CONTENT-TYPE="xml/">
<html>
  <head>
    
  </head>
  <body>
    <p>
      1. Item
    </p>
    <p>
      5. Item
    </p>
    <p>
      6. Item
    </p>
  </body>
</html>
</richcontent>
</node>
```

**Comment by leehach** [comment]
OK, changing the Content Type seems to have done the trick. All of my old maps, which were created before the Markdown editor was added, automatically loaded with the Content Type as Standard.

Follow-up questions:

1. Is there any way to have the note panel display in Markdown instead of formatted output, the same way that the Standard mode lets you switch to the HTML Code view panel?
2. When I double-click to edit a Markdown note (or edit the note in a dialog) is there anyway to get a word wrapped view instead of having long lines hang off the right side of the panel/window?

**Reply by macmarrum** [reply]
I should mention another option for exporting a map to markdown, i.e. by using a script. Here's one example:
- https://github.com/freeplane/freeplane/issues/333

> 1. Is there any way to have the note panel display in Markdown instead of formatted output, the same way that the Standard mode lets you switch to the HTML Code view panel?

I only know `Edit->Note->Switch Note/Map Focus`, i.e. the edit mode, so what you described would be a new feature request.

> 2. When I double-click to edit a Markdown note (or edit the note in a dialog) is there anyway to get a word wrapped view instead of having long lines hang off the right side of the panel/window?

It would be nice to have a word-wrap feature in the editor. It would be useful also for Groovy scripts and LaTeX – the same code editor is used in all 3 cases. Please feel free to create a feature request.

**Comment by leehach** [comment]
Thanks. I added feature request #614 and #615.
